packages/flutter/test/widgets/autocomplete_test.dart

```diff
@@ -421,6 +421,109 @@ void main() {
     expect(textEditingController.text, lastOptions.elementAt(0));
   });
 
+  group('optionsViewOpenDirection', () {
+    testWidgets('unset (default behavior): open downward', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Center(
+              child: RawAutocomplete<String>(
+                optionsBuilder: (_) => <String>['a'],
+                fieldViewBuilder: (_, TextEditingController controller, FocusNode focusNode, __) {
+                  return TextField(controller: controller, focusNode: focusNode);
+                },
+                optionsViewBuilder: (_, __, ___) => const Text('a'))))));
+      await tester.showKeyboard(find.byType(TextField));
+      expect(tester.getBottomLeft(find.byType(TextField)),
+        offsetMoreOrLessEquals(tester.getTopLeft(find.text('a'))));
+    });
+
+    testWidgets('down: open downward', (WidgetTester tester) async {
+      await tester.pumpWidget(
+          MaterialApp(
+            home: Scaffold(
+              body: Center(
+                child: RawAutocomplete<String>(
+                  optionsViewOpenDirection: AutocompleteOptionsViewOpenDirection.down, // ignore: avoid_redundant_argument_values
+                  optionsBuilder: (_) => <String>['a'],
+                  fieldViewBuilder: (_, TextEditingController controller, FocusNode focusNode, __) {
+                    return TextField(controller: controller, focusNode: focusNode);
+                  },
+                  optionsViewBuilder: (_, __, ___) => const Text('a'))))));
+      await tester.showKeyboard(find.byType(TextField));
+      expect(tester.getBottomLeft(find.byType(TextField)),
+        offsetMoreOrLessEquals(tester.getTopLeft(find.text('a'))));
+    });
+
+    testWidgets('up: open upward', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Center(
+              child: RawAutocomplete<String>(
+                optionsViewOpenDirection: AutocompleteOptionsViewOpenDirection.up,
```

Comment:
Hmm yeah, with the "Autocomplete" prefix on the name of the enum, this gets kind of long.

I think it's OK to leave that prefix off, and make it just `OptionsViewOpenDirection`. My reasoning is:

* Those other types around it that have that prefix are typedefs, not enums. So there's basically no cost in the niceness of the API from making their names long, because calling code won't typically mention them.

* On the other side, the risk from having a less specific name is that it effectively lives in a single shared namespace with the other public names in the framework, and so it might collide in the future with some name we want elsewhere, or sound confusingly like something that'd appear elsewhere in the framework.

  But I think "options view open direction" is already specific enough that that's not a big worry: if there's something else that might be thought of as an "options view" and it has an "open direction", then I think it'd have to have enough in common with `RawAutocomplete` that basically the same set of values for the "open direction" will make sense for that other thing as for `RawAutocomplete`. Namely the "options view" can go above, or go below, or better yet in the future would be able to automatically go where it fits but the app should decide whether to prefer above or below.

  So if there ever is such a collision, I think we can cheerfully use the same enum for both. We'd just revise its docs to be worded more generally, and to mention both uses under "See also:".

Reply:
That reasoning makes sense; will do.